### ISSUE TITLE
Endless ranges crash mutant

### ISSUE THREAD

**Issue by dgollahon** [mbj/mutant#1026]
Simple reproduction case I tried:

```ruby
class Test
  def self.slice_after(array, index)
    array[index..]
  end
end
```

Reproduction spec:

```ruby
RSpec.describe Test do
  it 'works' do
    expect(described_class.slice_after([1,2], 1)).to eql([2])
  end
end
```

Executing `mutant --use rspec Test` produces a long trace ending in
> 1: from /Users/gollahon/.rvm/gems/ruby-2.6.6/gems/mutant-0.9.9/lib/mutant/mutator/node.rb:42:in `mutate_child'
/Users/gollahon/.rvm/gems/ruby-2.6.6/gems/mutant-0.9.9/lib/mutant/mutator.rb:18:in `mutate': undefined method `type' for nil:NilClass (NoMethodError)

Versions
`mutant-rspec` version: 0.9.9
`unparser` version: 0.4.7

**Comment by mbj** [comment]
@dgollahon thanks for the report. This indicates I need to find more code bases to hook up to mutants self test in generation only mode to assert syntax coverage.

I think I should also hook up all code snippets from `parser` to mutants generator test suite.

**Comment by dgollahon** [comment]
> I think I should also hook up all code snippets from parser to mutants generator test suite.

Yeah, I think that would be a great idea.

**Comment by chaubunhtha** [comment]
.md

**Comment by mbj** [comment]
@chaubunhtha you are good? Seems you post random one word comments to this repository for no reason.

Unless you start to explain yourself, I've to block you assuming you are just here to spam.

**Comment by mbj** [comment]
@dgollahon fix for this issue was released in 0.9.10.

**Comment by dgollahon** [comment]
@mbj, thanks!
